file/writable_file_writer.h

```diff
@@ -112,6 +110,13 @@ class WritableFileWriter {
 #else  // !ROCKSDB_LITE
     (void)listeners;
 #endif
+    if (file_checksum_gen_factory != nullptr) {
+      FileChecksumGenOptions checksum_gen_options;
```

Comment:
I think this should be called `FileChecksumGenContext` following compaction filter's naming convention.